web/pgadmin/static/js/helpers/ModalProvider.jsx

```diff
@@ -100,13 +100,50 @@ function confirm(title, text, onOkClick, onCancelClick, okLabel = gettext('Yes')
       onCancelClick?.();
       closeModal();
     };
+
     const onOkClickClose = () => {
       onOkClick?.();
       closeModal();
     };
-    const extraButtons =  extras?.(closeModal);
+
+    return (
+      <AlertContent text={text} confirm onOkClick={onOkClickClose} onCancelClick={onCancelClickClose} okLabel={okLabel} cancelLabel={cancelLabel} extraButtons={extras?.(closeModal)} />
+    );
+  });
+}
+
+function confirmDelete(title, text, onOkClick, onCancelClick, okLabel = gettext('Yes'), cancelLabel = gettext('No'), type = null){
+  // bind the modal provider before calling
+  this.showModal(title, (closeModal) => {
```

Comment:
this code duplicates confirm function. Shouldn't we call confirm inside confirmDelete with extras?